src/sign-in-blazorserver/README.md

```diff
@@ -0,0 +1,112 @@
+| **Name**                        | `active-directory-dotnet-blazorserver`                              | Suggested value for this sample. <br/> You can change the app name at any time. |
+| **Supported account types**     | **Accounts in this organizational directory only (Single tenant)**  | Suggested value for this sample.                                                |
+| **Platform type**               | **Web**                                                             | Required value for this sample                                                  |
+| **Redirect URIs**               | `http://localhost:5000/signin-oidc`                                 | Required value for this sample                                                  |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Open the project in your code editor
+
+Next, open the _Api.csproj_ project under the the sign-in-blazorserver folder in your code editor.
+
+### 3. Update code sample in _appsettings.json_ with app registration values
+
+Finally, set the following values in _appsettings.json_.
+
+```json
+"TenantId": "[Enter 'common', or 'organizations' or the Tenant ID (Obtained from the Azure portal. Select 'Endpoints' from the 'App registrations' blade and use the GUID in any of the URLs), e.g. da41245a5-11b3-996c-00a8-4d99re19f292]",
+"ClientId": "[Enter the Client Id (Application ID obtained from the Azure portal), e.g. ba74781c2-53c2-442a-97c2-3d60re42f403]",
+"ClientSecret": "[Copy the client secret added to the app from the Azure portal]",
+```
+
+## Run the application
+
+### 1. Run the webapp
+
+1. Execute the following command to get the app up and running:
+
+   ```bash
+   dotnet run
+   ```
+
+### 2. Signin into the Blazor Server application
+
+1. Once the web app is listening, navigate to https://localhost:5001. Click the Sign In button and complete the login process.
+
+   ![A screenshot of a Blazor Server application application guiding the user to click the "Sign In" button.](./app-signed-in.png)
+
+1. Click Sign Out
+
+   ![A screenshot of a Blazor Server application application guiding the user to click the "Sign Out" button.](./app-signed-out.png)
+
+
+## About the code
+
+This ASP.NET Core 6 Blazor Server application is created using .NET Blazor Server. The app is requiring the user to be authenticated in Azure Active Directory.
+
+When users navigates the home page, the application initiates an authentication flow using the Microsoft Authentication Library (MSAL). The user completes this flow in their default web browser. Upon successful authentication, an HTTP GET request to the Microsoft Graph /me endpoint is issued with the user's access token in the HTTP header. The response from the GET request is then displayed to the user.
```

Comment:
> Upon successful authentication, an HTTP GET request to the Microsoft Graph /me endpoint is issued with the user's access token in the HTTP header.

This might happen under the covers, but I don't see a raw HTTP GET in the code. Instead, I see a `Microsoft.Identity.Web` reference:

```csharp
using var response = await downstreamWebApi.CallWebApiForUserAsync("GraphApi").ConfigureAwait(false);
```

Also, I don't see a mention of M.I.W. anywhere in the README, which seems pretty important to include. Please take a crack at beefing up this section with the sample's relevant code and architecture details.

As a reminder, here's what's in the comment of the README template for the **About the code** section:

```Markdown
## About the code

<!-- REQUIRED: Short description (3-5 sentences) of the application's -->
<!--           architecture and the code that enables it to function. -->

  <!-- DO discuss ARCHITECTURE and CODE, *not* just what the app *does* (it's features).          -->
  <!-- DO NOT simply restate or slightly rephrase the README's intro paragraph.                   -->
  <!-- DO mention any authNZ LIBRARY used by the app.                                             -->
  <!-- DO discuss the primary LIBRARY TYPES used; how and WHY were they used?                     -->
  <!-- DO explain things like you were talking to a colleague - a fellow software engineer.       -->

<!-- OPTIONAL: DIAGRAM (image or ASCII) showing application architecture and/or authNZ flow. -->
```

For example, you could say it uses M.I.W., the client init is in _Program.cs_, and the code handling sign-in is in _Index.razor_.

The goal of this section is for you to give people a way to know:

- What the app is doing under the hood
- What the app is using to do it (libraries, types, token grant flows)
- Where in the project all the above happens and, most importantly

..._without them having to sift through the code in the project's files to find out_.

For the same reason we try to write the clean, portable, well-documented, and maintainable code, _real_ goal, of course, is so you never, ever have to hear about it again.

In short, **DOC IT INTO SILENCE**.

Reply:
Thought of another way to say it - let me know if this makes sense...

**Answer the question, "What is it DOING?" rather than "What does it DO?"**

Reply:
good observations and feedback @mmacy 

done | addressed them mostly from 4f5d9768f64b9f3eae8f2dbdf402c3080c0d341f, c7d7fa3cd94db12e819c58ea667e0d1f2766462a, 729eac4573b6810024e8b00a13974919543a2a10, and 92490b23de73faf6298e0a0c42020eaf76cd9684